tests/test_policies.py

```diff
@@ -161,8 +162,13 @@ def test_policy(env_name, policy_name, extra_overrides):
     for key in batch:
         batch[key] = batch[key].to(DEVICE, non_blocking=True)
 
-    # Test updating the policy
+    # Test updating the policy (and test that it does not mutate the batch)
+    batch_ = deepcopy(batch)
     policy.forward(batch)
+    assert set(batch) == set(batch_), "Batch keys are not the same after a forward pass."
+    assert all(
+        torch.equal(batch[k], batch_[k]) for k in batch
+    ), "Batch values are not the same after a forward pass."
```

Comment:
nice! Could we add the same test for backward ?

I am wondering if we could add the same test for torch.compile when it's ready
cc @marinabar

Reply:
Sorry but how could the backward pass mutate the batch?

Reply:
You are right

+ we decided to not work on torch.compile yet